drivers/video/video_arducam_mega.c

```diff
@@ -0,0 +1,1397 @@
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->ev, dev, VIDEO_CID_AUTO_EXPOSURE_BIAS,
+			      (struct video_ctrl_range){.min = 0, .max = 6, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->whitebal, dev, VIDEO_CID_WHITE_BALANCE_TEMPERATURE,
+			      (struct video_ctrl_range){.min = 0, .max = 4, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->colorfx, dev, VIDEO_CID_COLORFX,
+			      (struct video_ctrl_range){.min = 0, .max = 14, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->exp_auto, dev, VIDEO_CID_EXPOSURE_AUTO,
+			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->gain_auto, dev, VIDEO_CID_AUTOGAIN,
+			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(&ctrls->whitebalauto, dev, VIDEO_CID_AUTO_WHITE_BALANCE,
+			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	if (drv_data->features & MEGA_HAS_SHARPNESS) {
+		ret = video_init_ctrl(
+			&ctrls->sharpness, dev, VIDEO_CID_SHARPNESS,
+			(struct video_ctrl_range){.min = 0, .max = 8, .step = 1, .def = 0});
+		if (ret < 0) {
+			return ret;
+		}
+	}
+	ret = video_init_ctrl(
+		&ctrls->gain, dev, VIDEO_CID_GAIN,
+		(struct video_ctrl_range){.min = 0, .max = 1023, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(
+		&ctrls->exposure, dev, VIDEO_CID_EXPOSURE,
+		(struct video_ctrl_range){.min = 0, .max = 30000, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(
+		&ctrls->quality, dev, VIDEO_CID_JPEG_COMPRESSION_QUALITY,
+		(struct video_ctrl_range){.min = 0, .max = 65535, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(
+		&ctrls->lowpower, dev, VIDEO_CID_ARDUCAM_LOWPOWER,
+		(struct video_ctrl_range){.min = 0, .max = 65535, .step = 1, .def = 0});
+	if (ret < 0) {
+		return ret;
+	}
+	if (drv_data->features & MEGA_HAS_FOCUS) {
+		ret = video_init_ctrl(
+			&ctrls->focus_auto, dev, VIDEO_CID_FOCUS_AUTO,
+			(struct video_ctrl_range){.min = 0, .max = 65535, .step = 1, .def = 0});
+		if (ret < 0) {
+			return ret;
+		}
+	}
+	/* Read only controls */
+	ret = video_init_ctrl(
+		&ctrls->support_resolution, dev, VIDEO_CID_ARDUCAM_SUPP_RES,
+		(struct video_ctrl_range){.min = 0, .max = 65535, .step = 1, .def = 0});
+	ctrls->support_resolution.flags |= VIDEO_CTRL_FLAG_READ_ONLY;
+	if (ret < 0) {
+		return ret;
+	}
+	ret = video_init_ctrl(
+		&ctrls->support_special_effects, dev, VIDEO_CID_ARDUCAM_SUPP_SP_EFF,
+		(struct video_ctrl_range){.min = 0, .max = 65535, .step = 1, .def = 0});
+	ctrls->support_special_effects.flags |= VIDEO_CTRL_FLAG_READ_ONLY;
+	if (ret < 0) {
+		return ret;
+	}
```

Comment:
It might be possible to remove this CID as this information is already communicated by the presence/absence of `VIDEO_CID_COLORFX`.